### ISSUE TITLE
Rule Proposal: Warn if function used for filter() or map() does not have a return

### ISSUE THREAD

**Issue by alecxe** [alecxe/eslint-plugin-protractor#20]
Invalid usage:

``` javascript
element.all(by.css('.items li')).filter(function(elem, index) {
  elem.getText().then(function(text) {
    return text === 'Third';
  });
}).first().click();
```

Valid usage:

``` javascript
element.all(by.css('.items li')).filter(function(elem, index) {
  return elem.getText().then(function(text) {
    return text === 'Third';
  });
}).first().click();
```

This needs more real-life samples. Search through SO?..

**Comment by alecxe** [comment]
This is quite close to the built-in http://eslint.org/docs/rules/array-callback-return.

**Comment by alecxe** [comment]
Fixed by e47ec5186737fb58c6339fadf7924d6b6bc6f984.
